Tests/UIKitTests/UIButtonExtensionsTests.swift

```diff
@@ -144,35 +144,93 @@ final class UIButtonExtensionsTests: XCTestCase {
         XCTAssertEqual(button.titleForSelected, title)
     }
 
-    func testCenterTextAndImage() {
+    func testSetImageLeftText() {
```

Comment:
```Swift
    func testSetImageLeftText() {
        let button = UIButton()
        let image = UIImage(color: .green, size: CGSize(width: 10, height: 10))
        button.setTitleForAllStates("Title")
        button.setImageForAllStates(image)
        button.titleLabel?.font = UIFont.systemFont(ofSize: 10)

        XCTAssertNotNil(button.imageView)
        XCTAssertNotNil(button.titleLabel)

        let spacing: CGFloat = 20
        button.setImageLeftOfText(spacing: spacing)
        let imageFrame = button.imageView!.frame
        let titleFrame = button.titleLabel!.frame

        XCTAssertEqual(titleFrame.midY, imageFrame.midY, accuracy: 1.0)
        XCTAssertEqual(titleFrame.minX - spacing, imageFrame.maxX, accuracy: 1.0)
        XCTAssertEqual(imageFrame.minX, button.bounds.minX, accuracy: 1.0)
        XCTAssertEqual(titleFrame.maxX, button.bounds.maxX, accuracy: 1.0)
    }
```

Reply:
I think it should be changed to：
```
XCTAssert(imageFrame.minX >= button.bounds.minX)
XCTAssert(titleFrame.maxX <= button.bounds.maxX)
```

Reply:
> I think it should be changed to：
> 
> ```
> XCTAssert(imageFrame.minX >= button.bounds.minX)
> XCTAssert(titleFrame.maxX <= button.bounds.maxX)
> ```

Why? We're forcing the image and title insets, so they should be exact.

Reply:
Yes, our test should first meet the actual project, and then test. If sizeToFit() is used, it will always be established under unit testing, which is meaningless.

This is with sizeToFit:
<img width="337" alt="sizeToFit" src="https://user-images.githubusercontent.com/12928482/98622473-9507d980-2344-11eb-8b0b-7913e3c231c1.png">

And this is without sizeToFit:
<img width="300" alt="without" src="https://user-images.githubusercontent.com/12928482/98622512-ab159a00-2344-11eb-9821-50072bfd356b.png">

In actual projects this is what we want.


```
XCTAssert(imageFrame.minX >= button.bounds.minX)
XCTAssert(titleFrame.maxX <= button.bounds.maxX)
```
Their meaning is to detect whether the text and image exceed the bounds of the button.

Reply:
If the button lays out its size automatically, either using `sizeToFit` or using autolayout, the tests should be as I have defined them with exact edges. That's what `UIButton` does by default and it's what it should still do using these functions. If you want to manually set the frame size, which I strongly recommend against, then these conditions would be different. But we want to test for the "standard" scenario which is where the buttons are used "as-is", rather than with custom frames.

Reply:
Unit testing can be done your way.